usagov/usagov-2021
# USAGOV-1296 & USAGOV-1332:  Error messaging and restyling the Report an issue page

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1296
https://cm-jira.usa.gov/browse/USAGOV-1332

## Description


The "[Report an issue](https://www.usa.gov/site-issue-report-form)" page is currently very different from other forms on our website (Example: [Find and contact elected officials page](https://www.usa.gov/elected-officials)). This PR changes the way error messages are displayed on the form and a complete restyling of the page. 

**Code Changes:**

- Change the form properties such as: font size, padding, margin, alignment, etc. The new properties are equal to the properties of the form "[Find and contact elected officials](https://www.usa.gov/elected-officials))",
- Added gray background and center the button.
- Refactored and documented the form script (reportAProblem.js) so the code is easier to understand and maintain.
- Added new style classes to the scss to change form properties.
- Change error messages style (field error outline, error text style, etc)
- Added an alert box at the top of the form to show errors in the form.
- Delete “Please complete the fields below.” (keep in bold “All fields are required.”) for consistency between forms.


## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


1. Go to localhost/site-issue-report-form and localhost/es/reporte-problemas-en-este-sitio-web and check that it looks like this and that the content is correct:

English Version:
<img width="439" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/6e1936fb-d91a-43ad-8a22-a3a7cfec563e">


Spanish Version:
<img width="435" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/88caba3e-f450-4149-81dd-147e441e0ad5">


2. In both languages, leave all fields blank and submit the form. You should see the following error messages:

English Version:
<img width="333" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/9e6d39bb-6f18-49e1-912c-b4c44f45c7be">

Spanish Version:
<img width="330" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/a40029a6-605b-4d4c-9098-220f639824ef">

3. In both languages, fill out the fields one by one, in any order, and check that the error messages are only displayed in the fields that are empty. For the email, it should show an error if the text in that field is not an email or is blank. It should look something like this:

English Version:
<img width="331" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/6cd429cf-2086-4279-bdd1-1830668f879b">

Spanish Version:
<img width="328" alt="image" src="https://github.com/usagov/usagov-2021/assets/39020809/c1d0f72e-ff31-4ab3-81af-63da04704590">

### Requires New Config

### Requires New Content

### Validation Steps

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
6. Find the commit of this pull request.
7. Build and deploy the changes.
8. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
